Source/Core/DolphinWX/MainToolBar.cpp

```diff
@@ -0,0 +1,255 @@
+       {IDM_CONFIG_CONTROLLERS, TOOLBAR_CONTROLLER}}};
+
+  for (const auto& entry : bitmap_entries)
+    ApplyBitmap(entry.first, entry.second);
+
+  // Separate, as the play button is dual-state and doesn't have a fixed bitmap.
+  RefreshPlayButton();
+}
+
+void MainToolBar::ApplyDebuggerBitmaps()
+{
+  constexpr std::array<std::pair<int, ToolBarBitmapID>, 6> bitmap_entries{
+      {{IDM_STEP, TOOLBAR_DEBUG_STEP},
+       {IDM_STEPOVER, TOOLBAR_DEBUG_STEPOVER},
+       {IDM_STEPOUT, TOOLBAR_DEBUG_STEPOUT},
+       {IDM_SKIP, TOOLBAR_DEBUG_SKIP},
+       {IDM_GOTOPC, TOOLBAR_DEBUG_GOTOPC},
+       {IDM_SETPC, TOOLBAR_DEBUG_SETPC}}};
+
+  for (const auto& entry : bitmap_entries)
+    ApplyBitmap(entry.first, entry.second);
+}
+
+void MainToolBar::ApplyBitmap(int tool_id, ToolBarBitmapID bitmap_id)
+{
+  const auto& bitmap = m_icon_bitmaps[bitmap_id];
+
+  SetToolDisabledBitmap(tool_id, WxUtils::CreateDisabledButtonBitmap(bitmap));
+  SetToolNormalBitmap(tool_id, bitmap);
+}
+
+void MainToolBar::AddToolBarButtons()
+{
+  if (m_type == ToolBarType::Debug)
+  {
+    AddDebuggerToolBarButtons();
+    AddSeparator();
+  }
+
+  AddMainToolBarButtons();
+}
+
+void MainToolBar::AddMainToolBarButtons()
+{
+  AddToolBarButton(wxID_OPEN, _("Open"), TOOLBAR_FILEOPEN, _("Open file..."));
+  AddToolBarButton(wxID_REFRESH, _("Refresh"), TOOLBAR_REFRESH, _("Refresh game list"));
+  AddSeparator();
+  AddToolBarButton(IDM_PLAY, _("Play"), TOOLBAR_PLAY, _("Play"));
+  AddToolBarButton(IDM_STOP, _("Stop"), TOOLBAR_STOP, _("Stop"));
+  AddToolBarButton(IDM_TOGGLE_FULLSCREEN, _("FullScr"), TOOLBAR_FULLSCREEN, _("Toggle fullscreen"));
+  AddToolBarButton(IDM_SCREENSHOT, _("ScrShot"), TOOLBAR_SCREENSHOT, _("Take screenshot"));
+  AddSeparator();
+  AddToolBarButton(wxID_PREFERENCES, _("Config"), TOOLBAR_CONFIGMAIN, _("Configure..."));
+  AddToolBarButton(IDM_CONFIG_GFX_BACKEND, _("Graphics"), TOOLBAR_CONFIGGFX,
+                   _("Graphics settings"));
+  AddToolBarButton(IDM_CONFIG_CONTROLLERS, _("Controllers"), TOOLBAR_CONTROLLER,
+                   _("Controller settings"));
+}
+
+void MainToolBar::AddDebuggerToolBarButtons()
+{
+  AddToolBarButton(IDM_STEP, _("Step"), TOOLBAR_DEBUG_STEP, _("Step into the next instruction"));
+  AddToolBarButton(IDM_STEPOVER, _("Step Over"), TOOLBAR_DEBUG_STEPOVER,
+                   _("Step over the next instruction"));
+  AddToolBarButton(IDM_STEPOUT, _("Step Out"), TOOLBAR_DEBUG_STEPOUT,
+                   _("Step out of the current function"));
+  AddToolBarButton(IDM_SKIP, _("Skip"), TOOLBAR_DEBUG_SKIP,
+                   _("Skips the next instruction completely"));
+  AddSeparator();
+  AddToolBarButton(IDM_GOTOPC, _("Show PC"), TOOLBAR_DEBUG_GOTOPC,
+                   _("Go to the current instruction"));
+  AddToolBarButton(IDM_SETPC, _("Set PC"), TOOLBAR_DEBUG_SETPC, _("Set the current instruction"));
+}
+
+void MainToolBar::AddToolBarButton(int tool_id, const wxString& label, ToolBarBitmapID bitmap_id,
```

Comment:
The two methods above read a little weird with the "gap" between the label and the short help string. What do you think of reordering the parameters so that the two strings are next to each other?